### ISSUE TITLE
Enable building stream2 into a shared lib

### ISSUE THREAD

**Issue by GDYendell** [dectris/documentation#3]
Currently attempting to build a shared library against stream2 gives errors like:
```
relocation R_X86_64_32S against `.rodata' can not be used when making a shared object; recompile with -fPIC
```

Adding the compiler flag to the stream2, cbor and compression CMakeLists makes it work ([e.g.](https://github.com/dectris/documentation/commit/e4056e6e10210c47dccaab0ff40ef70c976013fe)), but I am not sure what the side effects of this would be.

It would be great to be able to build a shared library against main, however that may be implemented.

**Comment by kalcutter** [comment]
@GDYendell It looks like the CMake-idiomatic solution is to simply do the following:
```
set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```
Setting this variable will recursively affect directories/dependent projects.

**Comment by GDYendell** [comment]
Yep that works. Thanks, @kalcutter! [e.g.](https://github.com/dls-controls/eiger-detector/blob/stream2/cpp/data/third_party/CMakeLists.txt)
